tidy3d/components/data/monitor_data.py

```diff
@@ -2098,6 +2099,78 @@ def renormalize_fields(self, proj_distance: float) -> FieldProjectionAngleData:
         # compute updated fields and their coordinates
         return self.make_renormalized_data(phase, proj_distance)
 
+class DirectivityData(FieldProjectionAngleData):
+    """Data associated with a :class:`.DirectivityMonitor`.
+
+    Example
+    -------
+    >>> from tidy3d import FieldProjectionAngleDataArray, FluxDataArray
+    >>> f = np.linspace(1e14, 2e14, 10)
+    >>> r = np.atleast_1d(5)
+    >>> theta = np.linspace(0, np.pi, 10)
+    >>> phi = np.linspace(0, 2*np.pi, 20)
+    >>> coords = dict(r=r, theta=theta, phi=phi, f=f)
+    >>> farfield_values = (1+1j) * np.random.random((len(r), len(theta), len(phi), len(f)))
+    >>> scalar_field = FieldProjectionAngleDataArray(farfield_values, coords=coords)
+    >>> flux_values =  np.random.random(len(f))
+    >>> flux_field = FluxDataArray(flux_values, coords=coords)
+    >>> monitor = DirectivityMonitor(
+    ...     center=(1,2,3), size=(2,2,2), freqs=f, name='n2f_monitor', phi=phi, theta=theta
+    ...     )
+    >>> data = DirectivityData(
+    ...     monitor=monitor, Er=scalar_field, Etheta=scalar_field, Ephi=scalar_field,
+    ...     Hr=scalar_field, Htheta=scalar_field, Hphi=scalar_field,
+            flux=flux_field
+    ...     projection_surfaces=monitor.projection_surfaces,
+    ...     )
+    """
+
+    monitor: DirectivityMonitor = pd.Field(
+        ...,
+        title="DirectivityMonitor",
+        description="DirectivityMonitor with an angle-based projection grid.",
+    )
+
+    flux: FluxDataArray = pd.Field(
+        ..., title="Flux", description="Flux values in the frequency-domain."
+    )
+
+
+    @property
+    def gain(self) -> xr.DataArray:
```

Comment:
It's not really `gain`, right? Gain has a few additional effects to be considered.